lib/ansible/modules/system/selogin.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/python
+
+# (c) 2017, Petr Lautrbach <plautrba@redhat.com>
+# Based on seport.py module (c) 2014, Dan Keder <dan.keder@gmail.com>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: selogin
+short_description: Manages linux user to SELinux user mapping
+description:
+     - Manages linux user to SELinux user mapping
+version_added: "2.8"
+options:
+  login:
+    description:
+      - a Linux user
+    required: true
+  seuser:
+    description:
+      - SELinux user name
+    required: true
+  selevel:
+    aliases: [ serange ]
+    description:
+      - MLS/MCS Security Range (MLS/MCS Systems only) SELinux Range for SELinux login mapping defaults to the SELinux user record range.
+    required: false
+    default: s0
+  state:
+    description:
+      - Desired mapping value.
+    required: true
+    default: present
+    choices: [ 'present', 'absent' ]
+  reload:
+    description:
+      - Reload SELinux policy after commit.
+    required: false
+    default: yes
+notes:
+   - The changes are persistent across reboots
+   - Not tested on any debian based system
+requirements: [ 'libselinux-python', 'policycoreutils-python' ]
```

Comment:
This was originally written for Red Hat Enterprise Linux 7. But there's no libselinux and policycoreutils python2 bindings in Red Hat Enterprise Linux 8 Beta, and they'll be dropped in Fedora 31 as well. The same comment can be applied to lines 266 and 269

Reply:
Are there replacement libraries we should mention here? (Or is it just `-python3`?)

Reply:
Python 3 versions are packaged in `python3-libselinux` and `python3-policycoreutils`. But the module needs just `selinux` and `seobject` python modules and version depends on the current interpret version. I don't know how to make it understandable and correct.

On lines 267 and 270 I'd probably use module names could be used instead of package names:

````
module.fail_json(msg=missing_required_lib("selinux"), exception=SELINUX_IMP_ERR)
module.fail_json(msg=missing_required_lib("seobject"), exception=SELINUX_IMP_ERR)
````

Reply:
I changed it to `libselinux` and `seobject provided by policycoreutils`